### PR TITLE
Fix typing of deprecated aliases

### PR DESCRIPTION
**Pull request by TheTripleV**
This fixes type hints in Pylance. The module `__getattr__` throws off its checking of what types are actually available.

This hides the `__getattr__` from type checkers

### CODE REVIEW

File: commands2/subsystem.py [hunk 1]
```diff
@@ -28,15 +28,16 @@ class Subsystem(Sendable):
...
-        instance = super().__new__(cls)
-        super().__init__(instance)
-        SendableRegistry.addLW(instance, cls.__name__, cls.__name__)
-        # add to the scheduler
-        from .commandscheduler import CommandScheduler
-
-        CommandScheduler.getInstance().registerSubsystem(instance)
-        return instance
+    if not TYPE_CHECKING:
+        def __new__(cls, *arg, **kwargs) -> "Subsystem":
```
Comment (auscompgeek) [inline commands2/subsystem.py:32]: This seems wrong. We should return `typing.Self` instead.

Comment (TheTripleV) [reply commands2/subsystem.py:32]: I agree. But typing.Self was added in 3.11 according to https://docs.python.org/3/library/typing.html#typing.Self

**Comment by TheTripleV** [comment]
Here's what all the typehints are before this PR
Screenshot 2024-01-17 at 11 53 12 PM

File: commands2/subsystem.py (hunk 1 above)
Comment (auscompgeek) [reply commands2/subsystem.py:33]: We're already using typing_extensions in command, so let's just use it here?

Comment (TheTripleV) [reply commands2/subsystem.py]: Ok, swapped it over

File: commands2/__init__.py [hunk 2]
```diff
@@ -98,18 +100,23 @@
...
+            import warnings
 
-        warnings.warn("CommandBase is deprecated", DeprecationWarning, stacklevel=2)
-        return Command
+            warnings.warn("CommandBase is deprecated", DeprecationWarning, stacklevel=2)
+            return Command
 
-    raise AttributeError("module {!r} has no attribute " "{!r}".format(__name__, attr))
+        raise AttributeError(
+            "module {!r} has no attribute " "{!r}".format(__name__, attr)
```
Comment (auscompgeek) [inline commands2/__init__.py:121]:
```suggestion
            f"module {__name__!r} has no attribute {attr!r}"
```

**Review by auscompgeek — APPROVED** [review]
LGTM with nit

**Pull request merged**